napari/_vispy/canvas.py

```diff
@@ -39,12 +93,30 @@ def __init__(self, *args, **kwargs) -> None:
         # using an lru_cache.
         self.max_texture_sizes = get_max_texture_sizes()
 
-        self.events.ignore_callback_errors = False
-        self.context.set_depth_func('lequal')
+        self.scene_canvas.events.ignore_callback_errors = False
+        self.scene_canvas.context.set_depth_func('lequal')
+
+        # Connecting events from SceneCanvas
+        self.scene_canvas.events.draw.connect(self.viewer.dims.enable_play)
+        self.scene_canvas.events.draw.connect(self.vispy_camera.on_draw)
+
+        self.scene_canvas.events.mouse_double_click.connect(
+            self.on_mouse_double_click
+        )
+        self.scene_canvas.events.mouse_move.connect(self.on_mouse_move)
+        self.scene_canvas.events.mouse_press.connect(self.on_mouse_press)
+        self.scene_canvas.events.mouse_release.connect(self.on_mouse_release)
+        self.scene_canvas.events.mouse_wheel.connect(self.on_mouse_wheel)
+        self.scene_canvas.events.resize.connect(self.on_resize)
+        self.scene_canvas.events.draw.connect(self.on_draw)
+        self.viewer.events.theme.connect(self._on_theme_change)
+        self.viewer.layers.events.reordered.connect(self._reorder_layers)
+        self.viewer.layers.events.removed.connect(self._remove_layer)
+        self.destroyed.connect(self._disconnect_theme)
```

Comment:
Note for reviewers: this is following the same patterns as `VispyLayer`s inside `_vispy/layers` and soon overlays in #4894.